clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -1164,6 +1168,34 @@ public void testHandleNormalLeaveGroupResponse() {
         assertTrue(leaveGroupFuture.succeeded());
     }
 
+    @Test
+    public void testHandleNormalLeaveGroupResponseAndTruncatedLeaveReason() {
+        MemberResponse memberResponse = new MemberResponse()
+                                            .setMemberId(memberId)
+                                            .setErrorCode(Errors.NONE.code());
+        LeaveGroupResponse leaveGroupResponse =
+            leaveGroupResponse(Collections.singletonList(memberResponse));
+        String leaveReason = "Very looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong leaveReason that is 271 characters long to make sure that length limit logic handles the scenario nicely";
+        setupCoordinator(RETRY_BACKOFF_MS, Integer.MAX_VALUE, Optional.empty());
+
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
```

Comment:
Instead of duplicating all this code, could we reuse `setupLeaveGroup`? We could perhaps adds an overload - e.g. `setupLeaveGroup(leaveGroupResponse, reason)`. What do you think?

Reply:
thanks, understood about `KafkaAdminClient.java`.

yeah, in my initial attempt I had `setupLeaveGroup(LeaveGroupResponse leaveGroupResponse, String leaveReason) {`
but I didn't like the assertion bit (as it hides the truncation) 

```
LeaveGroupRequestData leaveGroupRequest = ((LeaveGroupRequest) body).data();
return leaveGroupRequest.members().get(0).memberId().equals(memberId) &&
       leaveGroupRequest.members().get(0).reason().equals(leaveReason.substring(0, 255));
}, leaveGroupResponse);
```
however, maybe it's not that bad.
alternatively, we can pass `actualLeaveReason` and `expectedLeaveReason` but it looks even scarier. 
I can try the former approach and show you the result.

Reply:
Using `leaveReason` and `expectedLeaveReason` is OK for me as long as we keep `setupLeaveGroup(leaveGroupResponse)`.

Reply:
@dajac I've drafted the change for the `KafkaAdminClient` (you've mentioned it earlier, sorry overlooked), it feels that the method `truncateIfRequired` should be coverted to an util one.
I'm thinking of putting it in `org.apache.kafka.common.utils.Utils` but could you please give any suggestion if there's a more suitable place?